### ISSUE TITLE
Replace moving RIP with a solution always preserving correctness

### ISSUE THREAD

**Issue by asteinha** [google/safeside#74]
In Meltdown examples we currently move the instruction pointer in a signal handler. If there were some register spills, the correctness of the C++ execution model can be violated. We should look for a solution that maintains current functionality of our Meltdown examples while always preserving correctness.

**Comment by mmdriley** [comment]
Not to pile too many hopes on this issue, but as we consider moving to a new mechanism and interface for fault-recovery it could be nice to keep an eye on supporting fault _suppression_ using TSX as well, maybe through one of two swappable implementations.

**Comment by asteinha** [comment]
> Not to pile too many hopes on this issue, but as we consider moving to a new mechanism and interface for fault-recovery it could be nice to keep an eye on supporting fault _suppression_ using TSX as well, maybe through one of two swappable implementations.

That's true, however I don't see it as a priority, because it is not a portable solution - does not work on AMD and on older Intel CPUs - and it is not neccessary in any of our current examples. I'll just check the Meltdown-OF whether it would not work better with Intel TSX on the newest Intel CPUs. That one performs much better on AMD.
